object-cache.php

```diff
@@ -278,13 +114,20 @@ function wp_cache_fetch_all() {
  * @param int       $delay  Number of seconds to wait before invalidating the items.
  * @return bool             Returns TRUE on success or FALSE on failure.
  */
-function wp_cache_flush( $delay = 0 ) {	
-	$caller = array_shift( debug_backtrace( DEBUG_BACKTRACE_IGNORE_ARGS, 1 ) );
-	if ( ! defined( 'WP_CLI' ) || ! WP_CLI ) {
+function wp_cache_flush( $delay = 0 ) {
+	if ( version_compare( PHP_VERSION, '5.4.0', '>=' ) ) {
+		$caller = debug_backtrace( DEBUG_BACKTRACE_IGNORE_ARGS, 1 );
+		$caller = $caller[0];
+	} else {
+		$caller = debug_backtrace( DEBUG_BACKTRACE_IGNORE_ARGS );
+		$caller = $caller[0];
+	}
+
+	if ( 'cli' !== php_sapi_name() ) {
```

Comment:
Will this change allow `wp_cache_flush` to be called when running unit tests? WordPress core flushes the cache at the shutdown of every single test class of theirs which, when combined with this dropin, instantly kills a test suite.

Reply:
Added #11 and optionally #12 to address test issues.